src/tiles/tiles.ts

```diff
@@ -363,6 +364,18 @@ export function geometryToTileRanges<G extends GeoJSONGeometry>(
       return [boundingBoxToTileRange(geometry, zoom, metatile, referenceTileGrid)];
     case geometry instanceof Polygon:
       return polygonToTiles(geometry, zoom, metatile, referenceTileGrid);
+    case geometry instanceof GeometryCollection: {
+      // tile ranges may overlap for overlapping geometries inside geometry collection
+      const tileRanges = geometry.geometries
```

Comment:
This looks like it'll return overlapping ranges multiple times, we'll want to make sure overlapping is returned only once.

Reply:
after giving it some thought i agree on this one